tmt/base.py

```diff
@@ -3432,6 +3432,27 @@ def reshape(self, tests: list['tmt.steps.discover.TestOrigin']) -> bool:
 
         return False
 
+    # TODO: Make the str type-hint more narrow
+    def add_phase(self, step: Union[str, tmt.steps.Step], phase: tmt.steps.Phase) -> None:
+        """
+        Add a phase dynamically to the current plan.
+
+        :param step: The (future) step in which to add the phase.
+        :param phase: The phase to add to the step.
+        """
+        if isinstance(step, str):
+            if step not in tmt.steps.STEPS:
+                raise GeneralError(f"Tried to add to an unknown step: {step}")
+            step = getattr(self, step)
+        assert isinstance(step, tmt.steps.Step)
+        if step.plan != self:
+            raise GeneralError(
+                f"Tried to add to a step belonging to a different plan: {step.plan}"
+            )
+        # FIXME: Check that the current step/phase will be in the future
```

Comment:
"will be in the future" - I don't follow, the sentence seems incomplete.

Reply:
In the future w.r.t. to the current step/phase. E.g. if we are in a `prepare` step and we are trying to insert a `discover` then it should fail.